src/ui/common/src/reducers/workflowDagResults.ts

```diff
@@ -0,0 +1,54 @@
+// Before we fully migrate workflow result page, this store is
+// only used for operator and artifact details page.
+import { createSlice } from '@reduxjs/toolkit';
+
+import { handleGetWorkflowDagResult } from '../handlers/getWorkflowDagResult';
+import { DagResultResponse } from '../handlers/responses/dag';
+import { LoadingStatus, LoadingStatusEnum } from '../utils/shared';
+
+export interface WorkflowDagResultsState {
+  results: {
+    [id: string]: {
+      status: LoadingStatus;
+      result?: DagResultResponse;
+    };
+  };
+}
+
+const initialState: WorkflowDagResultsState = { results: {} };
+
+export const workflowDagResultsSlice = createSlice({
+  name: 'workflowDagResultsReducer',
+  initialState: initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(handleGetWorkflowDagResult.pending, (state, { meta }) => {
```

Comment:
We always add a pending, fulfilled, and rejected case for each handler. I wonder if it is worth it to create a helper function that creates these cases for us & puts the loading status to loading/succeeded/failed for us so all we need to do is provide function inputs of what else we would want to do in those cases.

Reply:
That's a good idea. The design wouldn't be trivial though since these pattern applies to different state object, and we require some level of customization to control these behaviors. It may require some work to get the typing right. Feel free to take a look at the current cases and add your suggestions!